aries-site/src/themes/scaled.js

```diff
@@ -1,26 +1,87 @@
-import { base } from 'grommet/themes';
+import { generate } from 'grommet/themes/base';
 import { deepMerge } from 'grommet/utils';
 import { aries } from './aries';
 
-export const scaled = (theme = aries, scale = 1) => {
-  const source = deepMerge(base, theme);
+export const scaled = (theme = aries, scale = 1, baseSpacing = 24) => {
+  // scales everything in base
+  const scaledBase = generate(baseSpacing * scale);
+  // merges theme, but overwrites scale for particular attributes
+  const source = deepMerge(scaledBase, theme);
   const scaledTheme = { ...source };
+  // define attributes which need to be re-scaled; essentially any
+  // size which is staticly set in hpe/aries theme
   const scaleAttributes = [
```

Comment:
whoa, this is a lot of items. I assume this has do be somewhat manual (which is probably fine since we won't be changing theme structure) -- but any thoughts on how to make sure this scales with any additions/changes to theme?

Reply:
I know. It's unfortunate, and the right question to ask. I am trying to balance forward progress (i.e. ability to get the hi-fidelity dashboards unblocked) with a reasonably sound approach. 

Right now, I think that balance is using a little brute force. Brute force in this case is essentially using the array as a "config file." In a way, our HPE-Theme is essentially a giant config file as well.

That said, the two approaches I have in mind are:
1. Make the hpe-theme dynamic, substituting any static pixel-based values with its own `baseSpacing` and calculation. But this would involve changing the theme and I don't think we are ready for that at this time. However, we shouldn't let go of this thought either as there are potential features we could enable such as the Gmail-like default, comfortable, and compact views which would allow end-users to customize their preferred settings.
2. Alternatively, I was thinking of recursively traversing the theme object and doing some sort of parsing, calculation, and substitution.
3. Lastly, this just came to mind, is that we could convert the theme object to a string, use regexp to target any string containing 'px', parse that, then replace with a string template literal containing our desired calculation.

None of these are the most elegant and will likely have edge cases where we would want to opt-out of the transformation, particularly in some of our `extend` cases implementing CSS directly.

As of now, I am thinking we need to move forward, but can continue thinking on this as I would eventually like to enable the "customize my layout" type functionality.

Reply:
I'm open to other ideas as well.

Reply:
Good thoughts. I definitely agree with the "let's move forward" mindset. I'll leave this comment open here to reference later on.

Reply:
FYI, the grommet-theme-designer has some of this theme manipulation built into it. It might be interesting to extract functions like this and ship them with grommet to allow re-use. https://github.com/grommet/grommet-theme-designer/blob/37c471971d734e45c0fc226567f51e5734d0f92f/src/Spacing.js#L5

Reply:
I think it would be a nice feature to have.